source and target data types (#584)
SimpleDataLabsInc/prophecy-docs: docs/Spark/gems/source-target/source-target.md

---
title: Source & Target
id: source-target

> description: Set of gems related to the input and output of data.

Reviewer: Okay this is something super small we can discuss - whether to use periods at the end of descriptions. Usually, long sentences do not fit on "preview" cards (look at the end of [this page](https://docs.prophecy.io/concepts/project/) to see what i'm talking about)
Therefore, I've been using shorter phrases without a period at the end. WDYT?

tags: []
---

> The following data types contain sets of gems to help you work with various file-based formats, connect to a warehouse-style data source, and work with various table-based formats.

Reviewer: ```suggestion
Prophecy supports a number of gems to help you work with various file-based formats, connect to a warehouse-style data source, and work with various table-based formats.
```

## File

> The file data type contains a collection of gems related to how you work with various file-based formats.

Reviewer: I wouldn't say it "contains" things. If you look in the product, source/target gems are not grouped by type. They can, however, be filtered by type. So I would say something more like
```suggestion
The following gems let you work with various file-based formats.
```

| [Parquet](./file/parquet)           | Parquet is an open source file format built to handle flat columnar storage data formats.                         |
| [Text](./file/text)                 | This gem allows you to read from or write to text file.                                                           |
| [XLSX (Excel)](./file/xlsx)         | Allows you to read or write Excel-compatible files.                                                               |
| [XML](./file/xml)                   | Allows you to read or write Excel-compatible files.                                                               |
| [Binary](./file/binary)             | Allows you to read or write Excel-compatible files.                                                               |
| [Hudi](./file/hudi)                 | Allows you to read or write Excel-compatible files.                                                               |

## Warehouse

> The warehouse data type contains a collection of gems that specialize in connecting to warehouse-style data sources.

Reviewer: same comment here and for catalog too
